src/mssqlProtocolHandler.ts

```diff
@@ -0,0 +1,267 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import * as Utils from "./models/utils";
+import { IConnectionInfo } from "vscode-mssql";
+
+enum Command {
+    connect = "/connect",
+    openConnectionDialog = "/openConnectionDialog",
+}
+
+/**
+ * Handles MSSQL protocol URIs.
+ */
+export class MssqlProtocolHandler {
+    constructor() {}
+
+    /**
+     * Handles the given URI and returns connection information if applicable. Examples of URIs handled:
+     * - vscode://ms-mssql.mssql/connect?server=myServer&database=dbName&user=sa&authenticationType=SqlLogin
+     * - vscode://ms-mssql.mssql/connect?connectionString=Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=myPassword;
+     *
+     * @param uri - The URI to handle.
+     * @returns The connection information or undefined if not applicable.
+     */
+    public handleUri(uri: vscode.Uri): IConnectionInfo | undefined {
```

Comment:
Can you add a test to validate that this checks and extracts the correct IConnectionInfo?